quinn-proto/src/cid_generator.rs

```diff
@@ -1,17 +1,20 @@
 use rand::RngCore;
+use std::time::Duration;
 
 use crate::shared::ConnectionId;
 use crate::MAX_CID_SIZE;
 
 /// Generates connection IDs for incoming connections
 pub trait ConnectionIdGenerator: Send {
-    /// Generates a new CID
+    /// Generates a new CID with finite lifetime
     ///
     /// Connection IDs MUST NOT contain any information that can be used by
     /// an external observer (that is, one that does not cooperate with the
     /// issuer) to correlate them with other connection IDs for the same
     /// connection.
-    fn generate_cid(&mut self) -> ConnectionId;
+    ///
+    /// Connection IDs will be proactively retired if lifetime is given a value.
+    fn generate_cid(&mut self) -> (ConnectionId, Option<Duration>);
```

Comment:
Good call. Since the cid interface change is not merged into master yet. It may be possible for me to add `Option<Duration>` directly there to avoid incompatibility.

Reply:
Updated